### PR TITLE
chore: update platform trigger

### PR DESCRIPTION
**Pull request by Vlukkee**
Update Platform triggers to do certain damage over time.
Add a float to the trigger to change the damage on the trigger options.

### CODE REVIEW

File: Game/Scripts/HealthSystem.cpp [hunk 1]
```diff
@@ -70,4 +70,9 @@ void HealthSystem::HealLife(float amountHealed)
 bool HealthSystem::EntityIsAlive() const
 {
 	return currentHealth > 0;
+}
+
+float HealthSystem::GetCurrentLife() const
```
Comment (MarcGonzalezMoratona) [inline Game/Scripts/HealthSystem.cpp:75]:
```suggestion
float HealthSystem::GetCurrentHealth() const
```

File: Game/Scripts/HealthSystem.h [hunk 2]
```diff
@@ -15,6 +15,7 @@ class HealthSystem : public Script
 
 	void TakeDamage(float damage);
 	void HealLife(float amountHealed);
+	float GetCurrentLife() const;
```
Comment (MarcGonzalezMoratona) [inline Game/Scripts/HealthSystem.h:18]:
```suggestion
	float GetCurrentHealth() const; 
```

File: Game/Scripts/UITrigger.cpp [hunk 3]
```diff
@@ -75,31 +76,39 @@ void UITrigger::Start()
...
+
 		if (isLoseTrigger == true)
 		{
 			uiGameStatesClass->LoseStateScene(true);
 		}
 		if (isLoseByDamage == true)
 		{
-			playerHealthSystem->TakeDamage(100);
+			playerHealthSystem->TakeDamage(damageTaken);
+			ENGINE_LOG("Taking damage %f --- Player Current Life %f", damageTaken, playerHealthSystem->GetCurrentLife());
```
Comment (MarcGonzalezMoratona) [inline Game/Scripts/UITrigger.cpp:93]:
```suggestion
			ENGINE_LOG("Taking damage %f --- Player Current Health %f", damageTaken, playerHealthSystem->GetCurrentHealth());
```

File: Game/Scripts/UITrigger.cpp [hunk 4]
```diff
@@ -75,31 +76,39 @@ void UITrigger::Start()
 
 void UITrigger::Update(float deltaTime)
 {
-
-}
-
-void UITrigger::OnCollisionEnter(ComponentRigidBody* other)
-{
-	if (other->GetOwner()->GetComponent(ComponentType::PLAYER))
+	if(onTriggerState != false)
```
Comment (MarcGonzalezMoratona) [inline Game/Scripts/UITrigger.cpp:79]: onTriggerState is a boolean?
Why aren't we just checking `if(onTriggerState)` or `ìf(!onTriggerState)`? It should do the same if the field is a boolean and it is more readable.

This also applies to other conditions of this file where we do `if(isWhateverTrigger == true)` , that can be refactored to `if(isWhateverTrigger)`

```suggestion
	if (onTriggerState)
```

File: Game/Scripts/UITrigger.cpp (hunk 3 above)
Comment (Dedado3000) [inline Game/Scripts/UITrigger.cpp:93]: why will a console log enter into develop? please erase it

**Review by Dedado3000 — CHANGES_REQUESTED** [review]
Can we change the name of the script?, why is called UITrigger

File: Game/Scripts/HealthSystem.cpp [hunk 5]
```diff
@@ -70,4 +70,9 @@ void HealthSystem::HealLife(float amountHealed)
 bool HealthSystem::EntityIsAlive() const
 {
 	return currentHealth > 0;
+}
+
+float HealthSystem::GetCurrenHealth() const
```
Comment (MarcGonzalezMoratona) [inline Game/Scripts/HealthSystem.cpp:75]:
```suggestion
float HealthSystem::GetCurrentHealth() const
```

File: Game/Scripts/UITrigger.cpp (hunk 3 above)
Comment (Vlukkee) [reply Game/Scripts/UITrigger.cpp]: Done!

File: Game/Scripts/UITrigger.cpp [hunk 6]
```diff
@@ -75,31 +76,39 @@ void UITrigger::Start()
...
 		{
 			uiGameStatesClass->WinStateScene(true);
 		}
-		
+
 		if (isLoseTrigger == true)
 		{
 			uiGameStatesClass->LoseStateScene(true);
 		}
 		if (isLoseByDamage == true)
```
[the remark this answers is not in the document]
Comment (Vlukkee) [reply Game/Scripts/UITrigger.cpp]: Done!

File: Game/Scripts/UITrigger.cpp (hunk 4 above)
Comment (Vlukkee) [reply Game/Scripts/UITrigger.cpp]: Done!

File: Game/Scripts/HealthSystem.h (hunk 2 above)
Comment (Vlukkee) [reply Game/Scripts/HealthSystem.h]: Nice! i dont saw it and its correlative with the all script. ty!

File: Game/Scripts/UITrigger.cpp [hunk 7]
```diff
@@ -75,31 +76,39 @@ void UITrigger::Start()
...
 		{
-			playerHealthSystem->TakeDamage(100);
+			playerHealthSystem->TakeDamage(damageTaken);
```
[the remark this answers is not in the document]
Comment (Vlukkee) [reply Game/Scripts/UITrigger.cpp]: Done!

File: Game/Scripts/HealthSystem.cpp (hunk 5 above)
Comment (Vlukkee) [reply Game/Scripts/HealthSystem.cpp]: Done!

**Review by Dedado3000 — DISMISSED** [review]

**Review by Dedado3000 — DISMISSED** [review]

**Review by Dedado3000 — APPROVED** [review]

**Review by diegusmejia117 — APPROVED** [review]

**Pull request merged**